File: 10/usb-example/python/test-peri.py
from practicum import find_mcu_boards
from peri import McuWithPeriBoard
from time import sleep

devs = find_mcu_boards()

if len(devs) == 0:
    print("*** No practicum board found.")
    exit(1)

b = McuWithPeriBoard(devs[0])
print("*** Practicum board found")
print("*** Manufacturer: %s" % \
        b.handle.getString(b.device.iManufacturer, 256))
print("*** Product: %s" % \
        b.handle.getString(b.device.iProduct, 256))

count = 0
while True:
    b.setLedValue(count)
    sw = b.getSwitch()
    light = b.getLight()

    if sw is True:
        state = "PRESSED"
    else:
        state = "RELEASED"

    print("LEDs set to %d | Switch state: %-8s | Light value: %d" % (
            count, state, light))

    count = (count + 1) % 8
    sleep(0.5)

File: 10/usb-example/java/src/practicum/TestPeri.java
package practicum;

import org.usb4java.Device;

public class TestPeri
{
    public static void main(String[] args) throws Exception
    {
        McuBoard.initUsb();

        try
        {
        	Device[] devices = McuBoard.findBoards();
        	
        	if (devices.length == 0) {
                System.out.format("** Practicum board not found **\n");
                return;
        	}
        	else {
                System.out.format("** Found %d practicum board(s) **\n", devices.length);
        	}
            McuWithPeriBoard peri = new McuWithPeriBoard(devices[0]);

            System.out.format("** Practicum board found **\n");
            System.out.format("** Manufacturer: %s\n", peri.getManufacturer());
            System.out.format("** Product: %s\n", peri.getProduct());

            int count = 0;

            while (true) 
            {
                Thread.sleep(500);
                peri.setLedValue(count);
                boolean sw = peri.getSwitch();
                int light = peri.getLight();
                System.out.format("LED set to %d | Switch state: %s | Light: %d\n",
                        count, sw, light);

                count++;
                if (count > 7) count = 0;
            }
        }
        catch (Exception e)
        {
            System.out.println(e);
        }

        McuBoard.cleanupUsb();
    }
}
